jujugui/static/gui/src/app/components/entity-content/entity-content.js

```diff
@@ -279,10 +279,17 @@ YUI.add('entity-content', function() {
     _showEntityRelations: function() {
       var entityModel = this.props.entityModel;
       if (entityModel.get('entityType') === 'charm') {
-        return (
-          <juju.components.EntityContentRelations
-            changeState={this.props.changeState}
-            relations={entityModel.get('relations')} />);
+        // Need to flatten out the relations to determine if we have any.
+        var relations = entityModel.get('relations');
+        var requires = relations.requires || {};
+        var provides = relations.provides || {};
+        var relationsList = Object.keys(requires).concat(Object.keys(provides));
+        if (relationsList.length > 0) {
```

Comment:
Instead of assigning empty objects above and then doing the concat couldn't you just do `if (relations.requires || relations.provides) {...` or can `relations.requires` or `relations.provides` sometimes return an empty object?